### PR TITLE
Hotfix/sequential batch

### PR DESCRIPTION
**Pull request by tabby-or-not**
- Only call sequential endpoint if more than one request
- Better handle errored requests, passing error response into data if data undefined

### CODE REVIEW

**Review by furious-luke — APPROVED** [review]

File: src/batch.js [hunk 1]
```diff
@@ -113,9 +113,11 @@ export default class Batch extends Middleware {
   }
 
   selectUrl = batch => {
-    for (const b of batch) {
-      if (this.mutMethods.has(b.request.method.toLowerCase())) {
-        return this.mutableBatchUrl
+    if (Object.keys(batch).length > 1) {
```
Comment (scaredcat) [inline src/batch.js]:
```suggestion
    if (batch.length > 1) {
```

**Review by scaredcat — APPROVED** [review]

File: src/batch.js [hunk 2]
```diff
@@ -137,31 +139,25 @@ export default class Batch extends Middleware {
...
+          data,
+        })
+      } else {
+        batch[ii].reject({
+          response: r,
+          data: data || {errors: {detail: r.reason_phrase}},
         })
-        results.push(r.body)
       }
+      results.push(data)
```
Comment (tabby-or-not) [inline src/batch.js:161]: @scaredcat Should we be consistent and pass the generated 'error data' from 157 into here too?

Comment (tabby-or-not) [reply src/batch.js:161]:
```
        if (!data) data = {errors: {detail: r.reason_phrase}}
        batch[ii].reject({
          response: r,
          data,
        })
```

Comment (scaredcat) [reply src/batch.js:161]: probably better to do it that way, yes.

**Pull request merged**